plugins/search/src/components/HomePageComponent/HomePageSearchBar.tsx

```diff
@@ -67,7 +60,7 @@ export const HomePageSearchBar = ({
 
   return (
     <SearchBarBase
-      className={className}
+      className={props.className ? props.className : classes.searchBar}
```

Comment:
Tbh I think this behavior is quite unexpected, typically you'd merge like we were doing previously. Perhaps a couple of `'unset'` or `'initial'` css values can help do this instead?

Reply:
Hmm interesting perspective, If I provide a className prop I would expect it to override the other one without me having to reset the default css values. (Would love to learn more about the recommendations for this if you can point to any official docs) 

In any event, I think it's something else going on here and I'm still looking into what is happening. 

The default styles keep overriding the custom styles if it's merged and not replaced. See attached screenshot
<img width="1440" alt="Screen Shot 2022-01-20 at 1 21 26 PM" src="https://user-images.githubusercontent.com/25153114/150338985-c5b96223-0112-49b2-9402-703327a17d76.png">
but the problem is that it sometimes works locally and sometimes not, and then when out in production, it does not override the css at all. Found some issues on import order of components on material-ui so looking into that

Reply:
Don't wanna scavenge all of the internet at this point x), but for example MUI takes the merge approach https://github.com/mui-org/material-ui/blob/ad2535a8b15f76355f4559dbac02e9cb379269ba/packages/mui-material/src/Avatar/Avatar.js#L180

Reply:
Hmm, that priority issue is strange idd though. Perhaps this is why MUI v5 switches out the style system? x)

Maybe you could set it up to be able to forward `classes` instead of `className`? I *think* that would be more robust. You'd need to add a `classes` prop that you forward to `useStyles`, https://v4.mui.com/styles/advanced/#overriding-styles-classes-prop

Reply:
@Rugvip Yes, that did seem to solve the issue! Please take a look!